R/develop-request.R

```diff
@@ -0,0 +1,218 @@
+#' Build a Google API request
+#'
+#' Intended primarily for internal use in client packages that provide
+#' high-level wrappers for users.
+#'
+#' @param endpoint List of information about the target endpoint. Presumably
+#' prepared from the [Discovery
+#' Document](https://developers.google.com/discovery/v1/getting_started#background-resources) for the target API.
+#' @param params Named list. Values destined for URL substitution, the query,
+#' or, for `develop_request()` only, the body.
+#' @param base_url Character.
+#' @param method Character. An HTTP verb, such as `GET` or `POST`.
+#' @param path Character. Path to the resource, not including API's `base_url`.
+#' Examples: `drive/v3/about` or `drive/v3/files/{fileId}`. If `path` includes
+#' variables inside curly brackets, these are substituted using named parameters
+#' found in `params` by `build_request()`.
+#' @param body List. Values to send in the API request body.
+#' @param key API key. Needed for requests that don't contain a token. For more,
+#'   see Google's document [Credentials, access, security, and
+#'   identity](https://support.google.com/googleapi/answer/6158857?hl=en&ref_topic=7013279).
+#'   A key can be passed as a named component of `params`, but note that the
+#'   formal argument `key` will clobber it, if non-`NULL`.
+#' @param token Token, ready for inclusion in a request, i.e. prepared with
+#'   [httr::config()].
+#'
+#' @section develop_request():
```

Comment:
Should this be in backticks?

Reply:
Yes, it should but I wrote this right after making the usethis website:

http://usethis.r-lib.org/reference/ci.html

Yes I should just backtick it and assume all the presentation formats will get sorted out.

Reply:
done